google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -0,0 +1,414 @@
+          .put(':', "%3A")
+          .put(';', "%3B")
+          .put('=', "%3D")
+          .put('?', "%3F")
+          .put('@', "%40")
+          .put('[', "%5B")
+          .put(']', "%5D")
+          // In addition to [a-zA-Z0-9], these chars should not be URI-encoded:
+          .put('-', "-")
+          .put('_', "_")
+          .put('.', ".")
+          .put('~', "~")
+          .build();
+
+  private static final String ACCOUNT = "account";
+  private static PrivateKey privateKey;
+  private static PublicKey publicKey;
+
+  private StorageOptions options;
+  private StorageRpcFactory rpcFactoryMock;
+  private StorageRpc storageRpcMock;
+  private Storage storage;
+
+  private Blob expectedBlob1, expectedBlob2, expectedBlob3, expectedUpdated;
+  private Bucket expectedBucket1, expectedBucket2, expectedBucket3;
+
+  @BeforeClass
+  public static void beforeClass() throws NoSuchAlgorithmException, InvalidKeySpecException {
+    KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+    EncodedKeySpec privateKeySpec =
+        new PKCS8EncodedKeySpec(BaseEncoding.base64().decode(PRIVATE_KEY_STRING));
+    privateKey = keyFactory.generatePrivate(privateKeySpec);
+    EncodedKeySpec publicKeySpec =
+        new X509EncodedKeySpec(BaseEncoding.base64().decode(PUBLIC_KEY_STRING));
+    publicKey = keyFactory.generatePublic(publicKeySpec);
+  }
+
+  @Before
+  public void setUp() {
+    rpcFactoryMock = mock(StorageRpcFactory.class);
+    storageRpcMock = mock(StorageRpc.class);
+    when(rpcFactoryMock.create(any(StorageOptions.class))).thenReturn(storageRpcMock);
+    options =
+        StorageOptions.newBuilder()
+            .setProjectId("projectId")
+            .setClock(TIME_SOURCE)
+            .setServiceRpcFactory(rpcFactoryMock)
+            .setRetrySettings(ServiceOptions.getNoRetrySettings())
+            .build();
+  }
+
+  private void initializeService() {
+    storage = options.getService();
+    initializeServiceDependentObjects();
+  }
+
+  private void initializeServiceDependentObjects() {
+    expectedBlob1 = new Blob(storage, new BlobInfo.BuilderImpl(BLOB_INFO1));
+    expectedBlob2 = new Blob(storage, new BlobInfo.BuilderImpl(BLOB_INFO2));
+    expectedBlob3 = new Blob(storage, new BlobInfo.BuilderImpl(BLOB_INFO3));
+    expectedBucket1 = new Bucket(storage, new BucketInfo.BuilderImpl(BUCKET_INFO1));
+    expectedBucket2 = new Bucket(storage, new BucketInfo.BuilderImpl(BUCKET_INFO2));
+    expectedBucket3 = new Bucket(storage, new BucketInfo.BuilderImpl(BUCKET_INFO3));
+    expectedUpdated = null;
+  }
+
+  @Test
+  public void testGetOptions() {
+    initializeService();
+    assertSame(options, storage.getOptions());
+  }
+
+  @Test
+  public void testCreateBucket() {
+    when(storageRpcMock.create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS))
+        .thenReturn(BUCKET_INFO1.toPb());
+    initializeService();
+    Bucket bucket = storage.create(BUCKET_INFO1);
+    assertEquals(expectedBucket1, bucket);
```

Comment:
Just an advice you can use the `verify()` method on the mock object to verify that the specified conditions are met.

`verify(storageRpcMock).create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS)`

Reply:
Thanks for that comment.  Checking that conditions are met is verified in 
`when(storageRpcMock.create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS))`
So, no need to verify that externally.
But what does worth to check is this method is called once.
I did that as:
`  when(...).thenReturn(...).thenThrow('fail);`